src/dotty/tools/dotc/transform/Erasure.scala

```diff
@@ -319,7 +355,7 @@ object Erasure extends TypeTestsCasts{
           assert(sym.isConstructor, s"${sym.showLocated}")
           select(qual, defn.ObjectClass.info.decl(sym.name).symbol)
         }
-        else if (qualIsPrimitive && !symIsPrimitive || qual.tpe.isErasedValueType)
+        else if (qualIsPrimitive && !symIsPrimitive || qual.tpe.widen.isErasedValueType)
```

Comment:
Are you sure? `widenDealias` for AnnotatedType calls `dealias` which does:
```scala
      case tp: AnnotatedType =>
        tp.derivedAnnotatedType(tp.annot, tp.tpe.dealias)
```
So the result should still be an AnnotatedType.